docs/guides/nextauth-migration.mdx

```diff
@@ -0,0 +1,253 @@
+```
+
+## Reading user and session data
+
+Replace any `getServerSession(req, res, authOptions)` to `getAuth(req)` from Clerk.
+
+```tsx
+// With NextAuth.js
+
+export async function getServerSideProps(context) {
+  const session = await getServerSession(context.req, context.res, authOptions)
+	...
+```
+
+```tsx
+// With Clerk
+
+export async function getServerSideProps(context) {
+  const session = getAuth(context.req);
+	...
+	return { props: { ...buildClerkProps(ctx.req) } };
+};
+```
+
+Replace NextAuth's `useSession` with Clerk's `useAuth` (or `useUser` if you need the full user object)
+
+```tsx
+// With NextAuth.js
+
+import { useSession } from "next-auth/react"
+...
+const { data: session } = useSession()
+```
+
+```tsx
+// With Clerk
+
+import { useAuth, useUser } from "@clerk/nextjs";
+...
+const { userId, sessionId, getToken } = useAuth();
+const { isSignedIn, user } = useUser();
+```
+
+**User IDs as Foreign Keys**       
+
+When you migrate user data from NextAuth's database to Clerk, Clerk generates new user IDs for each user. If you are using existing user IDs as foreign keys in your database (e.g. in a `user_id` column), you can save those IDs as the user's `externalId` in Clerk.
+
+This `externalId` can be included in the session token by [adding this customization](https://clerk.com/docs/backend-requests/making/custom-session-token) -
+`"userId": "{{user.external_id || user.id}}"`
+
+In your application, you can access this id like this:
+
+```tsx
+const { sessionClaims: { userId } } = getAuth(req)
+const { userId, sessionId, getToken } = useAuth();
+```
+
+, you'll get the `externalId` for the migrated users, and the Clerk-generated ID for users who sign up after the migration.
+
+Alternatively, after the data migration, you can update all the user IDs stored in your database as a foreign key to the new Clerk user IDs.
+
+You can read more about user IDs and user data migration [here](https://github.com/clerk/migration-script).
+
+## Production Config
+
+Every Clerk application has a Development and a Production environment. Before you start migrating user data, you can enable Clerk's production environment and migrate your users directly onto the production instance.
+
+*You can migrate a small set of users on the development instance for testing/staging.*
+
+1. In the Clerk Dashboard, **create a production instance** for your application.
+2. Configure DNS settings as instructed
+3. Enable all the OAuth providers you have credentials for
+4. Copy the OAuth credentials from your environment to the Clerk Dashboard.
+5. Update the callback URLs in your OAuth applications on each respective service to the new URLs shown on the Clerk Dashboard.
+
+## Migrating User Data
+
+This walkthrough will help you move user data from your existing database to Clerk.
+
+T*o retain the user data in your database for easy querying, check out the documentation on [data synchronization with webhooks](https://clerk.com/docs/users/sync-data).*
```

Comment:
```suggestion
*To retain the user data in your database for easy querying, check out the documentation on [data synchronization with webhooks](https://clerk.com/docs/users/sync-data).*
```

This should probably be a Callout. It definitely does not belong in this section -- this is not related to migration user data. Perhaps its own section below.

Reply:
Reading this again, either the whole line should not be italics or if its that important it should be a Callout. There is another line just above that this applies to as well.